src/main/java/info/solidsoft/mockito/java8/AssertionMatcher.java

```diff
@@ -49,10 +47,10 @@
 
     private static final LambdaAwareHandyReturnValues handyReturnValues = new LambdaAwareHandyReturnValues();
 
-    private final Consumer<T> consumer;
+    private final CheckedConsumer<T> consumer;
     private String errorMessage;
 
-    private AssertionMatcher(Consumer<T> consumer) {
+    private AssertionMatcher(CheckedConsumer<T> consumer) {
```

Comment:
The problem here is to break backward compatibility. All the tests passing a consumer extracted to a variable would fail. E.g.:

```
    @Test
    public void shouldAllowToUseConsumerFunctionBoundToVariable() {
        //given
        Consumer<Integer> consumer = i -> assertThat(i).isEqualTo(2);
        //when
        ts.fireTorpedo(2);
        //then
        verify(ts).fireTorpedo(assertArg(consumer));
    }
```
It's not probably very common, but I can imagine cases where to reduce a duplication a lambda is extracted to a field or constant.
There would a need to have another method `assertArgChecked(CheckedConsumer<T> consumer)`, but it requires duplication in the `AssertionMatcher` class which I don't like. The same is for `Predicate`.

Do you have an idea how to implement it without so great duplication (and keeping backward compatibility)?

Reply:
I can only see `assertArgChecked` route, since there's no way distinguish between a `Consumer` and a `CheckedConsumer` when there's just the lambda available.